eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/LegacyJpaEventStorageEngine.java

```diff
@@ -285,7 +286,11 @@ public MessageStream<EventMessage<?>> source(@Nonnull SourcingCondition conditio
 
     private MessageStream<EventMessage<?>> eventsForCriteria(SourcingCondition condition,
                                                              EventCriteria criterion) {
-        var aggregateIdentifier = resolveAggregateIdentifier(criterion.tags());
+        if (!(criterion instanceof TagEventCriteria)) {
+            throw new IllegalArgumentException(
+                    "While using LegacyJpaEventStorageEngine, only singular TagEventCriteria are supported");
+        }
```

Comment:
We may have method on Criteria interface like `singleOrThrow()` for that. What do you think?